simple-glue-transform/app.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+from typing import Any
+
+import aws_cdk as cdk
+import aws_cdk.aws_iam as iam
+from aws_cdk.aws_glue import CfnCrawler
+from aws_cdk.aws_glue_alpha import (Code, GlueVersion, JobExecutable,
+                                    JobLanguage, JobType)
+from aws_ddk_core.base import BaseStack
+from aws_ddk_core.pipelines import DataPipeline
+from aws_ddk_core.resources import S3Factory
+from aws_ddk_core.stages import GlueTransformStage, S3EventStage
+from constructs import Construct
+
+app = cdk.App()
+
+
+class DdkApplicationStack(BaseStack):
+    def __init__(
+        self, scope: Construct, id: str, environment_id: str, **kwargs: Any
+    ) -> None:
+        super().__init__(scope, id, environment_id, **kwargs)
+
+        bucket = S3Factory.bucket(
+            self,
+            id="bucket",
+            environment_id=environment_id,
+            event_bridge_enabled=True,
+        )
+
+        s3_stage = S3EventStage(
+            self,
+            id="s3-event-capture",
+            environment_id=self._environment_id,
+            event_names=["Object Created"],
+            bucket_name=bucket.bucket_name,
+            key_prefix="source",
+        )
+
+        glue_crawler_role = iam.Role(
+            self,
+            "glue-crawler-role",
+            assumed_by=iam.ServicePrincipal("glue.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSGlueServiceRole"
+                )
+            ],
+        )
+        glue_crawler_role.add_to_policy(
+            iam.PolicyStatement(
+                actions=["s3:*"],
+                resources=[bucket.bucket_arn, f"{bucket.bucket_arn}/*"],
+            )
+        )
+
+        glue_transform_stage = GlueTransformStage(
+            self,
+            id="glue-transform-simple",
+            environment_id=environment_id,
+            executable=JobExecutable.of(
+                glue_version=GlueVersion.V2_0,
+                language=JobLanguage.PYTHON,
+                script=Code.from_asset("src/sample-job.py"),
+                type=JobType.ETL,
+            ),
+            database_name="glue-database",
+            targets=CfnCrawler.TargetsProperty(
+                s3_targets=[
+                    CfnCrawler.S3TargetProperty(
+                        path=f"s3://{bucket.bucket_name}/target/"
+                    )
+                ]
+            ),
+            crawler_role=glue_crawler_role,
+            job_args={
+                "--S3_SOURCE_PATH": f"s3://{bucket.bucket_name}/source/",
```

Comment:
Tiny nitpick, but instead of the formatter string you could do:
```python
bucket.arn_for_objects("source/")
```
and
```python
bucket.arn_for_objects("target/")
```

Reply:
Love it, thanks.